### DISCUSSION TITLE
logica for ontologies

### DISCUSSION THREAD

**Discussion by wdduncan** [EvgSkv/logica#395] [Ideas]
@EvgSkv I've started a repo for experimenting using logica with ontologies. For now, I've dubbed it [onto-logica](https://github.com/wdduncan/onto-logica/tree/main). Here is [notebook](https://github.com/wdduncan/onto-logica/blob/main/notebooks/dental-materials-logica.ipynb) demonstrating some basic usage. 

Anyway, just making you aware of it. You are welcome to participate if you wish.

**Comment by EvgSkv** [comment]
@wdduncan thanks for sharing!

Looks like there was some difficulty with getting leafs of the tree. It was caused by using of predicate level aggregation rather than aggregating expression. Aggregating expressions should be used if getting an empty list (or any other empty aggregation result) is desired rather than having entry filtered.

There was also an error about `MagicalEntangle` because database wasn't extended with Logica functions. (Also there wasn't a function exposed in Logica to do enrichment, which I added.)

I've created CoLab addressing these issues:
https://colab.research.google.com/drive/1N9WVgn6qxeAggUk1tjrDJR3DE4Wl1tVB?usp=sharing

Leaf finding works there :-)

![image](https://github.com/user-attachments/assets/1a059f2e-bb55-49b7-ac8f-94778387196a)


Happy to collaborate further on this project!
